### ISSUE TITLE
Allow subsequent changes to workflows

### ISSUE THREAD

**Issue by matthias-ronge** [kitodo/kitodo-production#4840]
## Description

In Production 3, workflows were introduced. A production template with tasks is generated from the workflow, from which processes are generated, while the tasks are copied into the individual processes. These copies are technically necessary to record the status of the individual tasks in the process (for example, the status of correction tasks). If you were to change a workflow afterwards, the tasks of the processes would not be aligned, so it is not possible to change workflows afterwards.

It is desirable to be able to change workflows afterwards. There are a number of reasons for this:

 - small adjustments to script parameters
 - adding roles from another tenant
 - subsequent insertion of a task
   - this can also be at the end for completed processes, for example adding an archiving

This should result in a change in the tasks of the production templates, and tasks in the processes should be adapted. This can be a long-running task when many processes are affected.

## To do

 * [ ] Conception phase: Which changes to the workflow are allowed? In which state should processes be after the change, depending on their state before the change? Possible permitted changes could be:

   - change the parameters of an existing task
   - insert a new task at the beginning or linearly between two existing tasks
   - insert a new task linearly at the end. If it is an automatic task, it should also be started for processes that have already been completed. That can be very many.
   - delete a task. What happens if users in processes are currently working on this task?
   - change the order of tasks. What happens here with processes in the correction run if the target task of the correction (where the correction was requested) now precedes the task that is supposed to carry out the correction?
   - do several tasks in parallel that were previously organized differently. In which status do processes have to be set in which these tasks are currently being processed?
   - any changes? If not, how do you distinguish programmatically between allowed and not allowed changes, and how is the user guided here?

 * [ ] Implementation of the desired changes.

**Comment by stefanCCS** [comment]
I personally like this feature, but I also understand, that this is not an easy one.
In my opinion the "Conception phase" is the critical one and should be done with most care as possible.
Even, if I am not in the evelopment team, in my opinion there is a high effort needed to do this.

**Comment by m-kotzyba** [comment]
This issues was dicussed often and is still wanted by several institutons.
Can you pls give it the label: "development fund 2022"

**Comment by BartChris** [comment]
I opened a discussion here: https://github.com/kitodo/kitodo-production/discussions/5351

**Comment by BartChris** [comment]
I tried to summarize different aspects discussed in the community meeting in a Github discussion: https://github.com/kitodo/kitodo-production/discussions/5418

**Comment by michaelkubina** [comment]
Read through the github discussion #5418 , because this is a feature, where we certainly have a demand for. i would not even bother, if we simply could replace the workflow for certain documents in a project completely, by which i mean: delete all tasks in the process, insert the new ones from the current workflow.

While i would rather not touch processes that are not completed yet, i would have no hard feelings about replacing the tasks on a process that has been finished already (all steps done). The replaced workflow could then also have all steps done, which does not feel any different to the state before. But then i would be able to revert to a certain task (even a new one that has not been configured in the original workflow but was added in the new workflow) in order to make use of it.

Would love to see this as a proposal for the development fund 2027 again ;-)
